File: Sesion 10/Ejercicio/Ejercicio1 S10/ejercicio1.java
import java.util.ArrayList;
import java.util.Collections;

public class Main {

    public static <T extends Comparable<T>> void imprimirArreglo(ArrayList<T> arreglo) {
        for (T elemento : arreglo) {
            System.out.print(elemento + " ");
        }
        System.out.println();
    }

    public static <T extends Comparable<T>> void ordenarArreglo(ArrayList<T> arreglo) {
        int n = arreglo.size();
        for (int i = 0; i < n - 1; ++i) {
            int min_idx = i;
            for (int j = i + 1; j < n; ++j) {
                if (arreglo.get(j).compareTo(arreglo.get(min_idx)) < 0) {
                    min_idx = j;
                }
            }
            if (min_idx != i) {
                Collections.swap(arreglo, i, min_idx);
            }
        }
    }

    public static void main(String[] args) {
        ArrayList<Integer> arreglo_enteros = new ArrayList<>();
        arreglo_enteros.add(5);
        arreglo_enteros.add(2);
        arreglo_enteros.add(7);
        arreglo_enteros.add(3);
        arreglo_enteros.add(1);
        arreglo_enteros.add(9);

        System.out.println("Arreglo original:");
        imprimirArreglo(arreglo_enteros);

        ordenarArreglo(arreglo_enteros);
        System.out.println("Arreglo ordenado:");
        imprimirArreglo(arreglo_enteros);

        ArrayList<Double> arreglo_dobles = new ArrayList<>();
        arreglo_dobles.add(3.5);
        arreglo_dobles.add(1.2);
        arreglo_dobles.add(8.9);
        arreglo_dobles.add(2.3);
        arreglo_dobles.add(5.6);

        System.out.println("Arreglo original:");
        imprimirArreglo(arreglo_dobles);

        ordenarArreglo(arreglo_dobles);
        System.out.println("Arreglo ordenado:");
        imprimirArreglo(arreglo_dobles);
    }
}

File: Sesion 10/Ejercicio/Ejercicio1 S10/ejercicio1.py
def imprimir_arreglo(arreglo):
    for elemento in arreglo:
        print(elemento, end=" ")
    print()

def ordenar_arreglo(arreglo):
    n = len(arreglo)
    for i in range(n - 1):
        min_idx = i
        for j in range(i + 1, n):
            if arreglo[j] < arreglo[min_idx]:
                min_idx = j
        if min_idx != i:
            arreglo[i], arreglo[min_idx] = arreglo[min_idx], arreglo[i]


arreglo_enteros = [5, 2, 7, 3, 1, 9]
print("Arreglo original:")
imprimir_arreglo(arreglo_enteros)

ordenar_arreglo(arreglo_enteros)
print("Arreglo ordenado:")
imprimir_arreglo(arreglo_enteros)

arreglo_dobles = [3.5, 1.2, 8.9, 2.3, 5.6]
print("Arreglo original:")
imprimir_arreglo(arreglo_dobles)

ordenar_arreglo(arreglo_dobles)
print("Arreglo ordenado:")
imprimir_arreglo(arreglo_dobles)
